man/fabric.7.md

```diff
@@ -251,15 +251,58 @@ Users can enable or disable available providers through build configuration
 options.  See 'configure --help' for details.  In general, a specific provider
 can be controlled using the configure option '--enable-<provider_name>'.  For
 example, '--enable-udp' (or '--enable-udp=yes') will add the udp provider to the
-build.  To disable the provider, '--enable-udp=no' can be used.
+build.  To disable the provider, '--enable-udp=no' can be used.  To build the
+provider as a stand-alone dynamically loadable library (i.e. DL provider),
+'--enable-udp=dl' can be used.
 
 Providers can also be enable or disabled at run time using the FI_PROVIDER
 environment variable.  The FI_PROVIDER variable is set to a comma separated
 list of providers to include.  If the list begins with the '^' symbol, then
 the list will be negated.
 
   Example: To enable the udp and tcp providers only, set:
-	FI_PROVIDER="udp,tcp"
+	`FI_PROVIDER="udp,tcp"`
+
+By default, DL providers can be found under the `libfabric` subdirectory
+of the directory where `libfabric.so` is installed. This location is
+determined when libfabric is built and installed. Libfabric also tries to
+find DL providers following the naming convention `lib<prov_name>-fi.so`
+using the search path of 'ld.so'.
+
+The FI_PROVIDER_PATH variable can be used to change the location to search
+for DL providers and how to resolve conflict if multiple providers with the
+same name are found.
+
+In the simplest form, the FI_PROVIDER_PATH variable is set to a colon
+separated list of directories. For example:
+
+	FI_PROVIDER_PATH=/opt/libfabric:/opt/libfabric2
+
+By default, if multiple providers (including the built-in providers) with the
+same name are found, the first one with the highest version is active and all
+the others are hidden. This can be changed by setting the FI_PROVIDER_PATH
+variable to start with '@', which force the first one to be active regardless
+of the version. For example:
+
+	FI_PROVIDER_PATH=@/opt/libfabric:/opt/libfabric2
+
+The FI_PROVIDER_PATH variable can also specify preferred providers by supplying
+full paths to libraries instead of directories. A preferred provider takes
+precedence over other providers with the same name. The specification of a
+preferred provider must be prefixed with '+'. For example:
+
+	FI_PROVIDER_PATH=+/opt/libfabric2/libtcp-fi.so:/opt/libfabric:+/opt/libfabric2/libudp-fi.so
+
+If FI_PROVIDER_PATH is set, but no directory is supplied, the default
+provider directory is used. Some examples:
+
+	FI_PROVIDER_PATH=
+	FI_PROVIDER_PATH=@
+	FI_PROVIDER_PATH=+/opt/libfabric/libtcp-fi.so
+	FI_PROVIDER_PATH=@+/opt/libfabric/libtcp-fi.so
+
+When FI_PROVIDER_PATH is set, even if it is empty, no search is performed
+using the search path of 'ld.so'.
```

Comment:
This statement and the one above the examples seem to contradict each other.  What is the behavior if FI_PROVIDER_PATH is set, but empty? What's the difference between empty versus = "@"?

Reply:
I could have worded it better. Basically, there are two things that are "default":
(1) the default provider directory under <install-dir>/lib/libfabric;
(2) the search path of ld.so

When FI_PROVIDER_PATH is not set at all, both (1) and (2) are searched, but the searching (2) are only for those well-formed filennames.

When FI_PROVIDER_PATH is set, but doesn't have directory (i.e.,is empty, or only have @, or only have items starts with +), (1) is searched but not (2).

Both 'FI_PROVIDER_PATH=' and 'FI_PROVIDER_PATH=@' would cause (1) be searched, the difference is the ordering.